tests/uri_router_test.py

```diff
@@ -9,6 +9,24 @@
 pytest.importorskip('srwl_bl')
 
 
+def test_error_for_bots():
+    from pykern.pkunit import pkeq, pkexcept, pkre
+    from pykern.pkcollections import PKDict
+    from sirepo import srunit
+
+    fc = srunit.flask_client()
+    fc.sr_login_as_guest()
+
+    uri = '/get-application-data'
+    d = PKDict(simulationType='srw', method='NO SUCH METHOD')
+
+    r = fc.post(uri, json=d)
+    pkeq(200, r.status_code)
+
+    r = fc.post(uri, environ_base={'HTTP_USER_AGENT': 'SPIDER'}, json=d)
```

Comment:
Prefer PKDict and set it to python-requests, i.e. `PKDict(HTTP_USER_AGENT='python-requests/2.26.0')`. python-requests should be considered a spider. If it isn't modify is_spider to recognize it and return True.

It's probably good that werkzeug/* is not a spider, because that would break all our tests, I think.

Reply:
f04f83c

Oddly, python-requests is a bot according to ```robot_detection``` but not ```user_agents```.  I added it to a list for ```is_spider()``` to check